### ISSUE TITLE
migrate-account test failed

### ISSUE THREAD

**Issue by ibmp33** [0xEigenLabs/eigen-secret#88]
Now the simple tests have been passed as per the [Readme](https://github.com/0xEigenLabs/eigen-secret#more-tasks), the next step we need make sure the complex cases work well.

common setup:  deploy contracts, and register one token at least, can use gist scripts [here](https://gist.github.com/eigmax/a5e0289fc7df121284c081e2d64eb7cb).

## update-account test( update user signing key )
| id | operations | result | expected |  pass or not | fix | 
|---|---|---|---|---|---|
| 0 |  common setup;<br> alice deposit(s) 10; <br> alias deposit(s) 10; <br> alias deposit(s) 10; <br> Bob deposit(s) 10; <br> Bob deposit(s) 10; <br> Bob sends Alice 11| - | Alice's balance should be 41; <br>Bob's Balance should be 9 |  y | - |
| 1 |  case 1; <br>Alice updateAccount| - | Alice's balance should be 41; <br>Bob's Balance should be 9 |  y | - |
| 2 |  case 2; <br>Alice send(s) Bob 22| - | Alice's balance should be 19; <br>Bob's Balance should be 31 |  y | - |
| 3 |  case 3; <br>Bob withdraw(s) 31| - | Alice's balance should be 19; <br>Bob's Balance should be 0 |  y | - |
| 4 |  case 4; <br>Bob updateAccount| - | Alice's balance should be 19; <br>Bob's Balance should be 0 |  y | - |
| 5 |  case 5; <br>Bob deposit(s) 10; <br> Alice sends Bob 11 <br> Bob sends Alice 12| - | Alice's balance should be 20; <br>Bob's Balance should be 9 |  y | - |
| 6 |  case 6; <br>Alice withdraw(s) 20| - | Alice's balance should be 0; <br>Bob's Balance should be 0 |  y | - |

## migrate-account test( update user accountKey )
| id | operations | result | expected |  pass or not | fix | 
|---|---|---|---|---|---|
| 0 |  common setup;<br> alice deposit(s) 10; <br> alias deposit(s) 10; <br> alias deposit(s) 10; <br> Bob deposit(s) 10; <br> Bob deposit(s) 10; <br> Bob sends Alice 11| - | Alice's balance should be 41; <br>Bob's Balance should be 9 |  y | - |
| 1 |  case 1; <br>Alice migrateAccount| - | Alice's balance should be 41; <br>Bob's Balance should be 9 |  n | - |
| 2 |  case 2; <br>Alice send(s) Bob 22| - | Alice's balance should be 19; <br>Bob's Balance should be 31 |   | - |
| 3 |  case 3; <br>Bob withdraw(s) 31| - | Alice's balance should be 19; <br>Bob's Balance should be 0 |   | - |
| 4 |  case 4; <br>Bob migrateAccount| - | Alice's balance should be 19; <br>Bob's Balance should be 0 |   | - |
| 5 |  case 5; <br>Bob deposit(s) 10; <br> Alice sends Bob 11 <br> Bob sends Alice 12| - | Alice's balance should be 20; <br>Bob's Balance should be 9 |   | - |
| 6 |  case 6; <br>Alice withdraw(s) 20| - | Alice's balance should be 0; <br>Bob's Balance should be 0 |   | - |

Error:

1. error_1

```
Error: Assert Failed.
Error in template ForceEqualIfEnabled_325 line: 56
Error in template SMTVerifier_402 line: 134
Error in template Membership_403 line: 25
Error in template Account_404 line: 189
Error in template UpdateState_492 line: 69

    at /Users/ioio33/eigen-secret/circuits/main_update_state_js/witness_calculator.js:161:27
    at Array.forEach (<anonymous>)
    at WitnessCalculator._doCalculateWitness (/Users/ioio33/eigen-secret/circuits/main_update_state_js/witness_calculator.js:136:14)
    at WitnessCalculator.calculateWTNSBin (/Users/ioio33/eigen-secret/circuits/main_update_state_js/witness_calculator.js:212:20)
    at Function.updateState (/Users/ioio33/eigen-secret/core/dist/prover.js:15:55)
    at async SecretSDK.migrateAccount (/Users/ioio33/eigen-secret/sdk/dist/index.js:650:37)
    at async SimpleTaskDefinition.action (/Users/ioio33/eigen-secret/tasks/account.ts:103:33)
    at async Environment._runTaskDefinition (/Users/ioio33/eigen-secret/node_modules/hardhat/src/internal/core/runtime-environment.ts:330:14)
    at async Environment.run (/Users/ioio33/eigen-secret/node_modules/hardhat/src/internal/core/runtime-environment.ts:163:14)
    at async main (/Users/ioio33/eigen-secret/node_modules/hardhat/src/internal/cli/cli.ts:277:7)
```

**Comment by eigmax** [comment]
Plz provide the test input data/

**Comment by eigmax** [comment]
fixed by #93
